### PR TITLE
contrib/bat: new package (0.23.0)

### PR DESCRIPTION
**Pull request by anellie**

### CODE REVIEW

**Comment by q66** [comment]
this should be built with system curl, libgit2 and libssh2, the latter two need adding

**Comment by q66** [comment]
note that both of these libraries should go in main/

**Comment by anellie** [comment]
Added `http-parser` and `libgit2`, other 2 were already present. It seems like `bat` is refusing to use the system versions though; they don't show in `ldd`. Adding `LIBGIT2_NO_VENDOR` didn't help either

**Comment by nekopsykose** [comment]
(i didn't check too deeply why that happens (same thing goes for zstd and most sys crates!) but i assume it's something to do with bindgen checking if the vendored crate headers match up to some system thing)

**Comment by anellie** [comment]
Should be good now, `ldd` lists all the libraries it should I think. Thanks for the tip @nekopsykose, gonna keep that in mind for future packages

File: contrib/bat/template.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+
+def post_patch(self):
+    from cbuild.util import cargo
+
+    self.cargo.vendor()
+    cargo.setup_vendor(self)
+
+
+def post_install(self):
```
Comment (q66) [inline contrib/bat/template.py:33]: please make use of the new install_completion helper

File: contrib/bat/template.py [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    self.install_man(next(self.find("target/", "bat.1")))
+    self.install_license("LICENSE-APACHE")
```
Comment (q66) [inline contrib/bat/template.py:35]: LICENSE-APACHE should not be installed

**Comment by q66** [comment]
outside of the little license nit, i don't see anything bad anymore

**Pull request merged**